log/scope.go

```diff
@@ -315,3 +338,96 @@ func (s *Scope) SetLogCallers(logCallers bool) {
 func (s *Scope) GetLogCallers() bool {
 	return s.logCallers.Load().(bool)
 }
+
+// Copy makes a copy of s and returns a pointer to it.
+func (s *Scope) Copy() *Scope {
+	out := *s
+	out.context = copyStringInterfaceMap(s.context)
+	out.labels = copyStringInterfaceMap(s.labels)
+	return &out
+}
+
+// WithLabels adds a key-value pairs to the labels in s.
+func (s *Scope) WithLabels(kvlist ...interface{}) *Scope {
+	out := s.Copy()
+	if len(kvlist)%2 != 0 {
+		out.labels["WithLabels error"] = fmt.Sprintf("even number of parameters required, got %d", len(kvlist))
```

Comment:
Just curious what `zap` does here? I can't think of a better way to do things, just curious if someone from zap has

Reply:
they use fprintf :-D